### DISCUSSION TITLE
ChromaDB Similarity Search Not Applying to Entire Collection

### DISCUSSION THREAD

**Discussion by zelhaddioui** [spring-projects/spring-ai#1108] [Q&A]
Hello,

I am encountering an issue with the ChromaVectorStore implementation in Spring AI, similar to a problem I faced with Elasticsearch. When performing a similarity search, it seems that the search is not applied to the entire collection. As a result, only a subset of the expected results is returned.

Here is the configuration setup I am using:
```bash
@Bean
public ChromaApi chromaApi() {
    return new ChromaApi(chromaApiUrl, new RestTemplate());
}

@Bean
public VectorStore chromaVectorStore(EmbeddingModel embeddingModel, ChromaApi chromaApi) {
    return new ChromaVectorStore(embeddingModel, chromaApi, "rag", true);
}
```
When I perform a similarity search like this:

List<Document> similarDocuments = vectorStore.similaritySearch(request);

The search does not cover the entire collection, leading to incomplete results.

**Comment by MhMdRiyazath** [comment]
In the similarity search method, it returns a document object created by the Spring AI team. This object only tracks content, metadata, and embedding. If you want to return all the data in the domain model, you need to add additional variables inside the metadata to retrieve those values during the similarity search.
